src/base/memory_monitor_sr.cpp

```diff
@@ -69,28 +102,71 @@ namespace cubmem
     return allocated_size;
   }
 
-  std::string memory_monitor::make_tag_name (const char *file, const int line)
+  void memory_monitor::add_stat (char *ptr, const size_t size, const char *file, const int line)
+  {
+    std::string tag_name;
+    MMON_METAINFO metainfo;
+
+    // size should not be 0 because of MMON_ALLOC_META_SIZE
+    assert (size > 0);
+
+    metainfo.allocated_size = (uint64_t) size;
+    m_total_mem_usage += metainfo.allocated_size;
+
+    tag_name = make_tag_name (file, line);
+
+    std::unique_lock <std::mutex> tag_map_lock (m_tag_map_mutex);
```

Comment:
NIT:

m_tag_map_mutex protects both `m_tag_map` and `m_stat_map` but, the name of it means that it only cosider `m_tag_map`.

Reply:
I was already aware of the question you raised, but I hadn't thought of a suitable name to express integrated information. What do you think about renaming it to `m_map_mutex`?

Reply:
Good to me.